### PR TITLE
Improved docs for bridge `RegisterPublisher` configuration API

### PR DESCRIPTION
**Pull request by ramonsmits**
Related to:

- https://github.com/Particular/NServiceBus.Transport.Bridge/pull/265

### CODE REVIEW

File: nservicebus/bridge/configuration.md [hunk 1]
```diff
@@ -31,6 +31,12 @@ The result is duplicate subscriptions for any endpoint that subscribes to an eve
 
 snippet: register-publisher
 
+Legacy transports versions that use message driven pubsub require the fully assembly qualified type name value to be passed. Note that it is not required to pass the culture and public key as only type name, assembly name and assembly version are used filtering subscribers by the message drive pubsub feature.
```
Comment (dvdstelt) [inline nservicebus/bridge/configuration.md:34]:
```suggestion
Legacy transports versions that use message-driven pub/sub require the fully qualified assembly type name value to be passed. Note that it is not required to pass the culture and public key as only the type name, assembly name, and assembly version are used in filtering subscribers by the message drive pub/sub feature.
```

**Review by dvdstelt — APPROVED** [review]

File: Snippets/Bridge/Bridge_1/Configuration.cs [hunk 2]
```diff
@@ -84,13 +84,19 @@ public void RegisterPublishers()
...
             invoicing.RegisterPublisher<OrderShipped>("Shipping");
+            invoicing.RegisterPublisher("Messages.OrderPlaced", "Sales");
 
             asb.HasEndpoint(invoicing);
 
             #endregion
+
+            #region register-publisher-legacy
+
+            invoicing.RegisterPublisher("CreditApproved, CreditScoring.Messages, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null", "Sales");
```
Comment (andreasohlund) [inline Snippets/Bridge/Bridge_1/Configuration.cs:97]: Is `, Culture=neutral, PublicKeyToken=null` needed?

File: nservicebus/bridge/configuration.md [hunk 3]
```diff
@@ -31,6 +31,12 @@ The result is duplicate subscriptions for any endpoint that subscribes to an eve
 
 snippet: register-publisher
 
+Legacy transports versions that use message-driven pub/sub require the fully qualified assembly type name value to be passed. Note that it is not required to pass the culture and public key as only the type name, assembly name, and assembly version are used in filtering subscribers by the message drive pub/sub feature.
```
Comment (andreasohlund) [inline nservicebus/bridge/configuration.md:34]: Link to https://docs.particular.net/samples/pubsub/message-driven/ ?

Comment (andreasohlund) [inline nservicebus/bridge/configuration.md:34]:
```suggestion
Legacy transports versions that use message-driven pub/sub require the fully qualified assembly type name value to be passed. Note that passing the culture and public key is not needed as only the type name, assembly name, and assembly version are used in filtering subscribers by the message-driven pub/sub-feature.
```

File: Snippets/Bridge/Bridge_1/Configuration.cs (hunk 2 above)
Comment (ramonsmits) [reply Snippets/Bridge/Bridge_1/Configuration.cs:97]: Nope, funny that you mention it as yesterday I did a commit on the bridge that removed these if they match the default values :-)
```suggestion
            invoicing.RegisterPublisher("CreditApproved, CreditScoring.Messages, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null", "Sales");
            // or, drop Culture and PublicKeyToken which are ignored by the message driven pub/sub feature
            invoicing.RegisterPublisher("CreditApproved, CreditScoring.Messages, Version=1.0.0.0", "Sales");
            
```

Comment (ramonsmits) [reply Snippets/Bridge/Bridge_1/Configuration.cs:97]: @andreasohlund how about this?

File: nservicebus/bridge/configuration.md (hunk 3 above)
Comment (ramonsmits) [reply nservicebus/bridge/configuration.md:34]:
```suggestion
Legacy transports versions that use [message-driven pub/sub](/nservicebus/messaging/publish-subscribe/#mechanics-message-driven-persistence-based) require the fully qualified assembly type name value to be passed. Note that it is not required to pass the culture and public key as only the type name, assembly name, and assembly version are used in filtering subscribers by the message drive pub/sub feature.
```

**Review by andreasohlund — APPROVED** [review]

**Pull request merged**